rust/lance/src/dataset/sql.rs

```diff
@@ -74,9 +80,16 @@ impl SqlQueryBuilder {
                 row_addr,
             )),
         )?;
+
+        Self::register_functions(&ctx);
```

Comment:
this only registers the function in a SQL query. I think we want to probably always register it in our session context.

Reply:
We have a process-wide shared context that gets initialized in `lance_datafusion::exec::get_session_context`.  We should probably move this registration logic there and use that context here.